### PR TITLE
    refactor(platform-server): simplify `_render` method     

### PR DESCRIPTION
**Pull request by alan-agius4**
This commits simplifies the `_render` method by using async/await.

### CODE REVIEW

File: packages/platform-server/src/utils.ts [hunk 1]
```diff
@@ -50,66 +50,57 @@ function appendServerContextInfo(applicationRef: ApplicationRef) {
...
-    const pendingTasksPromise = pendingTasks.whenAllTasksComplete;
-    return Promise.allSettled([isStablePromise, pendingTasksPromise]).then(() => {
-      const platformState = platform.injector.get(PlatformState);
-
-      const asyncPromises: Promise<any>[] = [];
-
-      if (applicationRef.injector.get(IS_HYDRATION_DOM_REUSE_ENABLED, false)) {
-        annotateForHydration(applicationRef, platformState.getDocument());
-      }
+  const moduleOrApplicationRef = await bootstrapPromise;
```
Comment (AndrewKushnir) [inline packages/platform-server/src/utils.ts:56]: nit: we can consider "await"ing this promise outside of this fn and just pass the `ApplicationRef` here.

**Review by AndrewKushnir — APPROVED** [review]
Looks great, just a minor proposal on awaiting a bootstrap promise outside of the function.

File: packages/platform-server/src/utils.ts (hunk 1 above)
Comment (alan-agius4) [reply packages/platform-server/src/utils.ts:56]: Good call.

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit 8c3b92cfb3688b687f31ebe8024cd88676086931.

**Pull request closed without merging**